pca/utils/dependency_injection.py

```diff
@@ -74,3 +74,34 @@ def decorator(obj: t.Callable) -> t.Callable:
         obj.__scope_function = scope_type
         return obj
     return decorator
+
+
+class Inject:
+    """
+    A descriptor for injecting dependencies as properties
+    """
+
+    container: Container = None
+    _type: t.Type = None
+
+    def __init__(self, name: str = None, interface: t.Type = None, qualifier: t.Any = None):
+        self._name = name
```

Comment:
All those fields probably should be public. When you come to `inject` decorator, you will have to acquire them from the outside of the descriptor body.